workers/unity/Packages/com.improbable.gdk.querybasedinterest/Constraint.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        public static ComponentInterest.QueryConstraint Box(double xLength, double yLength, double zLength, Vector3 center)
+        {
+            var constraint = Default();
+            constraint.BoxConstraint = new ComponentInterest.BoxConstraint
+            {
+                Center = new Coordinates(center.x, center.y, center.z),
+                EdgeLength = new EdgeLength(xLength, yLength, zLength)
+            };
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint RelativeSphere(double radius)
+        {
+            var constraint = Default();
+            constraint.RelativeSphereConstraint = new ComponentInterest.RelativeSphereConstraint
+            {
+                Radius = radius
+            };
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint RelativeCylinder(double radius)
+        {
+            var constraint = Default();
+            constraint.RelativeCylinderConstraint = new ComponentInterest.RelativeCylinderConstraint
+            {
+                Radius = radius
+            };
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint RelativeBox(double xLength, double yLength, double zLength)
+        {
+            var constraint = Default();
+            constraint.RelativeBoxConstraint = new ComponentInterest.RelativeBoxConstraint
+            {
+                EdgeLength = new EdgeLength(xLength, yLength, zLength)
+            };
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint EntityId(EntityId entityId)
+        {
+            var constraint = Default();
+            constraint.EntityIdConstraint = entityId.Id;
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint Component<T>() where T : ISpatialComponentData
+        {
+            var constraint = Default();
+            constraint.ComponentConstraint = Dynamic.GetComponentId<T>();
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint Component(uint componentId)
+        {
+            var constraint = Default();
+            constraint.ComponentConstraint = componentId;
+            return constraint;
+        }
+
+        public static ComponentInterest.QueryConstraint All(params ComponentInterest.QueryConstraint[] constraints)
+        {
+            if (constraints.Length > 0)
+            {
+                return new ComponentInterest.QueryConstraint
+                {
+                    AndConstraint = constraints.ToList(),
+                    OrConstraint = EmptyList
+                };
+            }
+
+            return Default();
+        }
+
+        public static ComponentInterest.QueryConstraint Any(params ComponentInterest.QueryConstraint[] constraints)
+        {
+            if (constraints.Length > 0)
```

Comment:
You can just return the body of this if statement I think? 

If the length == 0 then the result ends up being the same as `Default()` 

Same for `All()` above.